src/client/mod.rs

```diff
@@ -95,7 +95,7 @@ impl Client {
     /// See the documentation on
     /// [`ClientOptions::parse`](options/struct.ClientOptions.html#method.parse) for more details.
     pub async fn with_uri_str(uri: &str) -> Result<Self> {
-        let options = ClientOptions::parse(uri).await?;
+        let options = ClientOptions::parse_uri(uri, None).await?;
```

Comment:
Since both the sync and async versions of `ClientOptions::parse` both are thin wrappers around `ClientOptions::parse_uri`, calling that directly from within the driver seems simpler than adding conditional compilation everywhere.